src/lib/ThumbnailsSidebar.js

```diff
@@ -153,20 +154,30 @@ class ThumbnailsSidebar {
      * @param {Object} currentListInfo - VirtualScroller info object which contains startOffset, endOffset, and the thumbnail elements
      * @return {void}
      */
-    generateThumbnailImages({ items, startOffset }) {
-        if (!isFinite(startOffset) || startOffset < 0) {
-            return;
-        }
-
+    generateThumbnailImages({ items }) {
         this.currentThumbnails = items;
 
-        items.forEach((thumbnailEl, index) => {
-            if (thumbnailEl.classList.contains(CLASS_BOX_PREVIEW_THUMBNAIL_IMAGE_LOADED)) {
-                return;
-            }
+        // Serially renders the thumbnails one by one as needed
+        this.renderNextThumbnailImage();
+    }
 
-            this.requestThumbnailImage(index + startOffset, thumbnailEl);
-        });
+    /**
+     * Requests the next thumbnail image that needs rendering
+     *
+     * @return {void}
+     */
+    renderNextThumbnailImage() {
+        // Iterates over the current thumbnails and requests rendering of the first
+        // thumbnail it encounters that does not have an image loaded
+        for (let i = 0; i < this.currentThumbnails.length; i++) {
```

Comment:
It looks like you could still use the iterator style here, since `i` isn't used for anything other than getting the thumbnail element.

```
const nextThumbnail = this.currentThumbnails.find(thumbnailEl => (
    !thumbnailEl.classList.contains(CLASS_BOX_PREVIEW_THUMBNAIL_IMAGE_LOADED))
);
const parsedPageNum = parseInt(thumbnailEl.dataset.bpPageNumb, 10);
this.requestThumbnailImage(parsedPageNum - 1, thumbnailEl);
```